sway/input/input-manager.c

```diff
@@ -175,7 +175,15 @@ static struct sway_input_device *input_sway_device_from_wlr(
 static bool input_has_seat_fallback_configuration(void) {
 	struct sway_seat *seat = NULL;
 	wl_list_for_each(seat, &server.input->seats, link) {
-		struct seat_config *seat_config = seat_get_config(seat);
+		struct seat_config *seat_config = NULL;
+		struct seat_config *sc = NULL;
+    	for (int i = 0; i < config->seat_configs->length; ++i ) {
+    		sc = config->seat_configs->items[i];
+    		if (strcmp(seat->wlr_seat->name, sc->name) == 0) {
+    			seat_config = sc;
+    			break;
+    		}
+    	}
```

Comment:
Why are we iterating the seat configs here? I thought that the seat->config should be used now.

Reply:
Because `input_manager_verify_fallback_seat()` runs before `seat_apply_config()` can set the pointer (according to a backtrace).

Attempting to use the pointer now will result in undefined behavior and a "soft crash", hence this is the only time we need to iterate of the seats by name.